champion.py

```diff
@@ -30,43 +43,61 @@ class BaseChampion:
         ...
     """
 
-    def __init__(self, champion_name: str):
-        self.initialize_base_stat(champion_name)
+    def __init__(self, champion_name: str, level: int):
+        assert level in np.arange(1, 19), "Champion level should be in the [1,18] range"
+        self.level = level
+        self.base_stats = ALL_CHAMPION_BASE_STAT[champion_name]
+        self.update_stat_from_level()
 
-    def initialize_base_stat(self, champion_name):
+    def update_stat_from_level(self):
         """Takes all the base stats from the input dictionary and create the corresponding attributes in the instance"""
-        self.__dict__.update(ALL_CHAMPION_BASE_STAT[champion_name])
+
+        def calculate_flat_stat_from_level(base: float, mean_growth_perlevel: float, level: int):
+            # Check Readme for formula detail
```

Comment:
Why not put links to all references we use in comments https://leagueoflegends.fandom.com/wiki/Champion_statistic#Growth_statistic_calculations and (below) https://leagueoflegends.fandom.com/wiki/Attack_speed#Bonus_attack_speed_per_level